setup/src/Magento/Setup/Model/CryptKeyGenerator.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Setup\Model;
+
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Config\ConfigOptionsListConstants;
+use Magento\Framework\Math\Random;
+
+/**
+ * Generates a crypt
+ * @api
+ */
+class CryptKeyGenerator implements CryptKeyGeneratorInterface
+{
+    /**
+     * @var Random
+     */
+    private $random;
+
+    public function __construct(Random $random = null)
+    {
+        $this->random = $random ?? ObjectManager::getInstance()->get(Random::class);
+    }
+
+    /**
+     * Generates & returns a string to be used as crypt key.
+     *
+     * The key length is not a parameter, but an implementation detail.
+     *
+     * @return string
+     *
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function generate()
```

Comment:
Again these functions can have the php7 return type deceleration such as `: type` http://php.net/manual/en/migration70.new-features.php#migration70.new-features.return-type-declarations